crates/l2/sequencer/proof_coordinator.rs

```diff
@@ -379,43 +373,53 @@ impl GenServer for ConnectionHandler {
 
 async fn handle_connection(
     state: &ProofCoordinatorState,
-    mut stream: TcpStream,
+    stream: Arc<TcpStream>,
 ) -> Result<(), ProofCoordinatorError> {
     let mut buffer = Vec::new();
-    stream.read_to_end(&mut buffer).await?;
-
-    let data: Result<ProofData, _> = serde_json::from_slice(&buffer);
-    match data {
-        Ok(ProofData::BatchRequest { commit_hash }) => {
-            if let Err(e) = handle_request(state, &mut stream, commit_hash).await {
-                error!("Failed to handle BatchRequest: {e}");
-            }
+    // TODO: This should be fixed in https://github.com/lambdaclass/ethrex/issues/3316
+    // (stream should not be wrapped in an Arc)
+    match Arc::try_unwrap(stream) {
+        Err(_) => {
+            error!("Unable to use stream");
```

Comment:
Also, we can try and print the contained error here, to help potential debugging efforts.